examples/ui.apps/src/content/jcr_root/apps/core-components-examples/components/version/PackageVersionHelper.java

```diff
@@ -0,0 +1,70 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2021 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+
+import java.util.List;
+
+import javax.jcr.Session;
+
+import org.apache.jackrabbit.vault.packaging.JcrPackage;
+import org.apache.jackrabbit.vault.packaging.JcrPackageManager;
+import org.apache.jackrabbit.vault.packaging.PackageProperties;
+import org.apache.jackrabbit.vault.packaging.Packaging;
+
+import com.adobe.cq.sightly.WCMUsePojo;
+import com.day.cq.commons.Version;
+
+/**
+ * Helper Use-Object for displaying the most recent version of a given package name that is installed on the AEM instance
+ *
+ * Usage:
+ *    <sly data-sly-use.version="${'PackageVersionHelper' @ packageName='my.package.name'}">${version.packageVersion}</sly>
+ */
+public class PackageVersionHelper extends WCMUsePojo {
+
+    public static final String ATTR_PACKAGE_NAME = "packageName";
+
+    private Version packageVersion = null;
+
+    @Override
+    public void activate() throws Exception {
+        String packageName = get(ATTR_PACKAGE_NAME, String.class);
+        if (packageName != null) {
+            Session session = getResourceResolver().adaptTo(Session.class);
+            Packaging packaging = getSlingScriptHelper().getService(Packaging.class);
+            if (session != null && packaging != null) {
+                JcrPackageManager jcrPackageManager = packaging.getPackageManager(session);
+                if (jcrPackageManager != null) {
+                    List<JcrPackage> packages = jcrPackageManager.listPackages();
+                    for (JcrPackage jcrPackage : packages) {
+                        if (packageName.equals(jcrPackage.getPackage().getProperty(PackageProperties.NAME_NAME))) {
```

Comment:
I would recommend working with the high-level API `PackageProperties` instead of directly acting upon the low-level properties. The relevant method is https://jackrabbit.apache.org/filevault/apidocs/org/apache/jackrabbit/vault/packaging/VaultPackage.html#getId()

Reply:
Since this heavily depends on access privileges, why not just returning the bundle-version? That should work independent of any session privileges...

Reply:
@kwin , there are no properties for package name and version exposed in the `PackageProperties` API. You're right about privileges though, I'll have a look into using the bundle version.

Reply:
@vladbailescu The name and version is wrapped in https://jackrabbit.apache.org/filevault/apidocs/org/apache/jackrabbit/vault/packaging/PackageId.html exposed from `PackageProperties`

Reply:
Ha! Totally missed that, I was under the impression the id is just a string. Nice to know. I refactored the helper to get the version from the bundle, to avoid privileges issues. It's probably good enough to know the Core Components bundle version for our needs.